website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,510 @@
+            });
+            self.sortState('alphaAsc');
+        };
+
+        self.sortAlphabeticalDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('alphaDesc');
+        };
+
+        self.sortDateAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('dateAsc');
+        };
+
+        self.sortDateDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('dateDesc');
+        };
+
+        // Sorts nodes depending on current sort state.
+        self.sortBySortState = function () {
+            switch (self.sortState()) {
+                case 'alphaAsc':
+                    self.sortAlphabeticalAscending();
+                    break;
+                case 'alphaDesc':
+                    self.sortAlphabeticalDescending();
+                    break;
+                case 'dateAsc':
+                    self.sortDateAscending();
+                    break;
+                default:
+                    self.sortDateDescending();
+            }
+            if (self.filter()) {
+                self.quickSearch();
+            }
+        };
+
+        // For xs screen
+        self.sortFieldGivenDirection = function(){
+            var directionSort = self.preSelectDirection();
+            self.sortState(self.fieldSort() + directionSort);
+            self.sortBySortState();
+        };
+
+        // For xs screen
+        self.sortDirectionGivenField = function() {
+            var fieldSort = self.preSelectField();
+            self.sortState(fieldSort + self.directionSort());
+            self.sortBySortState();
+        };
+
+        // When shifting to xs screen, tells which field to automatically display in select
+        self.preSelectField = function(){
+            return self.sortState().split(/[A-Z][a-z]+/g)[0];
+        };
+
+        // When shifting to xs screen, tells which direction to automatically highlight in select
+        self.preSelectDirection = function(){
+            return self.sortState().match(/[A-Z][a-z]+/g)[0];
+        };
+
+        // Colors sort asc/desc buttons either selected or not-selected
+        self.colorSortButtons = function (sort) {
+            if (self.sortState() === sort) {
+                return 'selected';
+            }
+            else {
+                return 'not-selected';
+            }
```

Comment:
Especially since this function is so close to the next, using the one-line form would help in scanning the difference between the two (`sortState` vs. `preSelectDirection`).